runtime/compiler/aarch64/codegen/StackCheckFailureSnippet.cpp

```diff
@@ -22,20 +22,83 @@
 
 #include "codegen/StackCheckFailureSnippet.hpp"
 
+#include "codegen/ARM64Instruction.hpp"
 #include "codegen/CodeGenerator.hpp"
+#include "codegen/GCStackAtlas.hpp"
+#include "codegen/Linkage.hpp"
+#include "codegen/Machine.hpp"
+#include "il/symbol/ResolvedMethodSymbol.hpp"
 
 uint8_t *
 TR::ARM64StackCheckFailureSnippet::emitSnippetBody()
    {
-   TR_UNIMPLEMENTED();
+   TR::ResolvedMethodSymbol *bodySymbol = cg()->comp()->getJittedMethodSymbol();
+   TR::Machine *machine = cg()->machine();
+   TR::SymbolReference *sofRef = cg()->comp()->getSymRefTab()->findOrCreateStackOverflowSymbolRef(bodySymbol);
 
-   uint8_t *buffer = cg()->getBinaryBufferCursor();
-   return buffer;
+   uint8_t *cursor = cg()->getBinaryBufferCursor();
+   uint8_t *returnLocation;
+
+   getSnippetLabel()->setCodeLocation(cursor);
+
+   // Pass cg()->getFrameSizeInBytes() to jitStackOverflow in register x9.
+   //
+   const TR::ARM64LinkageProperties &linkage = cg()->getLinkage()->getProperties();
+   uint32_t frameSize = cg()->getFrameSizeInBytes();
+
+   if (frameSize <= 0xffff)
+      {
+      // mov x9, #frameSize
+      *(int32_t *)cursor = 0xd2800009 | (frameSize << 5);
+      cursor += ARM64_INSTRUCTION_LENGTH;
+      }
+   else
+      {
+      TR_UNIMPLEMENTED();
+      }
+
+   // add J9SP, J9SP, x9
```

Comment:
Just to clarify, you're assuming that the prologue is buying the frame and this instruction is just setting it back prior to calling out to the SOC, correct?

Reply:
I wrote the code for moving `J9SP` in the same way as the arm32 version does:

```
   // add R7, R7, R11
   *(int32_t *)buffer = 0xe087700b;
   buffer+=4;
```
(R7 is J9SP in arm32, R11 holds the frameSize here)